internal-webapps/org.wso2.bfsi.consent.management.endpoint/src/main/java/org/wso2/bfsi/consent/management/endpoint/api/ConsentValidationEndpoint.java

```
@@ -113,8 +113,8 @@ public Response validate(@Context HttpServletRequest request, @Context HttpServl
         if (ConsentUtils.getConsentJWTPayloadValidatorConfigEnabled()) {
             try {
                 ConsentExtensionUtils.validateJWTSignatureWithPublicKey(payload, requestSignatureAlias);
-                requestData = new JSONObject(Objects
-                        .requireNonNull(CommonUtils.decodeRequestJWT(payload, ConsentExtensionConstants.BODY)).toJSONString());
+                requestData = new JSONObject(Objects.requireNonNull(CommonUtils
```

Can't we check for null and throw a controlled exception here instead of using requireNonNull?

---

Fixed in https://github.com/wso2-extensions/bfsi-consent-management/pull/8/commits/7c13892e0735c7d32a51565d4369ab1e9eddbc7e